File: app/src/main/java/com/frolo/muse/engine/AudioFxObserver.java
package com.frolo.muse.engine;

import com.frolo.muse.model.preset.Preset;
import com.frolo.muse.model.reverb.Reverb;


public interface AudioFxObserver {

    void onEnabled(AudioFx audioFx);

    void onDisabled(AudioFx audioFx);

    void onBandLevelChanged(AudioFx audioFx, short band, short level);

    void onPresetUsed(AudioFx audioFx, Preset preset);

    void onBassStrengthChanged(AudioFx audioFx, short strength);

    void onVirtualizerStrengthChanged(AudioFx audioFx, short strength);

    void onReverbUsed(AudioFx audioFx, Reverb reverb);

}

File: app/src/main/java/com/frolo/muse/engine/AudioFxObserver.kt
package com.frolo.muse.engine

import com.frolo.muse.model.preset.Preset


interface AudioFxObserver {
    fun onEnabled(audioFx: AudioFx)
    fun onDisabled(audioFx: AudioFx)
    fun onBandLevelChanged(audioFx: AudioFx, band: Short, level: Short)
    fun onPresetUsed(audioFx: AudioFx, preset: Preset)
    fun onBassStrengthChanged(audioFx: AudioFx, strength: Short)
    fun onVirtualizerStrengthChanged(audioFx: AudioFx, strength: Short)
    fun onPresetReverbUsed(audioFx: AudioFx, presetReverbIndex: Short)
}